event-dispatcher/src/main/java/io/sk8s/event/dispatcher/EventDispatchingHandler.java

```diff
@@ -41,30 +51,122 @@
 	// TODO: Change key to ObjectReference or similar
 	private final Map<String, XFunction> functions = new HashMap<>();
 
-	@Autowired // TODO: merge in here?
-	private JobLauncher launcher;
+	@Autowired
+	private FunctionDeployer deployer;
 
-	private final KubernetesClient kubernetesClient;
+	private volatile long scalingFrequencyMs = 0L; // Wait forever
 
-	public EventDispatchingHandler(KubernetesClient kubernetesClient) {
-		this.kubernetesClient = kubernetesClient;
-	}
+	private final Thread scalingThread = new Thread(new ScalingMonitor());
+
+	private final Object mutex = new Object();
```

Comment:
seems like there are simpler (and more efficient) ways to handle the rendezvous via `java.util.concurrent` types

Reply:
I'm far from a concurrency expert, but my primary goal here is the `wait`ing that can be woken up early. What `java.util.concurrent` primitive helps with that? Let's discuss on the call (if this stays at all, that is)